### DISCUSSION TITLE
Quarto and Jupyter Notebook

### DISCUSSION THREAD

**Discussion by kvten** [quarto-dev/quarto-cli#838] [Q&A] [answered]
Are there plans to make Quarto standalone? aka not rely on Jupyter notebook extensions

I have both installed in vscode but would prefer to only use Quarto since its better/more options of the 2. Only thing holding me back from removing Jupyter is running code chunks. I get error if/when Jupyter extension is disabled.

Thanks!

**Comment by jjallaire** [comment, marked as answer]
You might need to clarify your question a bit (as to desired vs. actual behavior). Currently running Python cells requires the Python extension but I don't think it requires the Jupyter extension. If there are some content types that require the VS Code Jupyter extension that isn't something we can influence.

If on the other hand you are saying "can quarto execute python code without Jupyter" the answer is no (as Juptyer defines a protocol for code to generate rich embeddable output -- if we didn't use Jupyter we'd need to develop our own such protocol, which would be a huge undertaking that's fundamentally unnecessary given that Jupyter already exists).

**Comment by kvten** [comment]
Thanks for answering my question! My question was the 2nd one.

Just for more context, the error message `command jupyter.execSelectionInteractive' not found` would show up whenever I try to "Run Cell" in a qmd file (python code) with 1 or more of the following VS Code extensions: 1) _Jupyter_, 2) _Jupyter Keymap_, or 3) _Jupyter Notebook Renderers_ disabled.

**Reply by jjallaire** [reply]
Yes, that is indeed a service provided by the Jupiter extension. It would be quite a bit of effort for us to create an entire Python interactive window so I think for the time being this will be a requirement.
